packages/gatsby/cache-dir/navigation.js

```diff
@@ -97,10 +97,12 @@ const navigate = (to, options = {}) => {
   })
 }
 
-function shouldUpdateScroll(prevRouterProps, { location: { pathname } }) {
+function shouldUpdateScroll(prevRouterProps, routerProps) {
+  const { location: { pathname, hash } } = routerProps
   const results = apiRunner(`shouldUpdateScroll`, {
     prevRouterProps,
-    pathname,
```

Comment:
I think that if we leave `pathname` here for backward compatibility, this change would potentially could go in